File: windows/QMK Toolbox/Usb/Bootloader/AtmelDfuDevice.cs
using System.Threading.Tasks;

namespace QMK_Toolbox.Usb.Bootloader
{
    class AtmelDfuDevice : BootloaderDevice
    {
        public AtmelDfuDevice(UsbDevice d) : base(d)
        {
            if (d.RevisionBcd == 0x0936)
            {
                Type = BootloaderType.QmkDfu;
                Name = "QMK DFU";
            }
            else
            {
                Type = BootloaderType.AtmelDfu;
                Name = "Atmel DFU";
            }
            PreferredDriver = "WinUSB";
            IsEepromFlashable = true;
            IsResettable = true;
        }

        public async override Task Flash(string mcu, string file)
        {
            await RunProcessAsync("dfu-programmer.exe", $"{mcu} erase --force");
            await Task.Delay(5);
            await RunProcessAsync("dfu-programmer.exe", $"{mcu} flash --force \"{file}\"");
            await Task.Delay(5);
            await RunProcessAsync("dfu-programmer.exe", $"{mcu} reset");
        }

        public async override Task FlashEeprom(string mcu, string file)
        {
            if (Type == BootloaderType.AtmelDfu)
            {
                await RunProcessAsync("dfu-programmer.exe", $"{mcu} erase --force");
            }

            await RunProcessAsync("dfu-programmer.exe", $"{mcu} flash --force --suppress-validation --eeprom \"{file}\"");

            if (Type == BootloaderType.AtmelDfu)
            {
                PrintMessage("Please reflash device with firmware now", MessageType.Bootloader);
            }
        }

        public async override Task Reset(string mcu) => await RunProcessAsync("dfu-programmer.exe", $"{mcu} reset");
    }
}

File: linux/src/qmk_toolbox/bootloader/atmel_dfu_device.py
from .bootloader_device import BootloaderDevice
from .bootloader_type import BootloaderType


class AtmelDfuDevice(BootloaderDevice):
    def __init__(self, usb_device):
        super().__init__(usb_device)
        self.bootloader_type = BootloaderType.ATMEL_DFU
        self.name = "Atmel DFU"
        self.preferred_driver = "usbhid"
        self.is_eeprom_flashable = True
        self.is_resettable = True
    
    async def flash(self, mcu: str, file_path: str) -> int:
        result = await self._run_process("dfu-programmer", [mcu, "erase", "--force"])
        if result != 0:
            return result
        
        result = await self._run_process("dfu-programmer", [mcu, "flash", file_path])
        if result != 0:
            return result
        
        return await self._run_process("dfu-programmer", [mcu, "reset"])
    
    async def flash_eeprom(self, mcu: str, file_path: str) -> int:
        result = await self._run_process("dfu-programmer", [mcu, "eeprom", file_path])
        if result != 0:
            return result
        return await self._run_process("dfu-programmer", [mcu, "reset"])
    
    async def reset(self, mcu: str) -> int:
        return await self._run_process("dfu-programmer", [mcu, "reset"])
